PayPal/src/main/java/com/braintreepayments/api/PayPalClient.java

```diff
@@ -113,12 +126,33 @@ public void onResult(@Nullable final Configuration configuration, @Nullable Exce
                     callback.onResult(manifestInvalidError);
                     return;
                 }
+                String clientId = configuration.getPayPalClientId() != null ? configuration.getPayPalClientId() : "Client ID Is Null";
+                String redirectUrl = getPayPalReturnUrl(activity);
+                Environment environment = Environment.SANDBOX;
+                if ("production".equalsIgnoreCase(configuration.getEnvironment())) {
+                    environment = Environment.LIVE;
+                }
+
+                CheckoutConfig checkoutConfig = new CheckoutConfig(
+                    activity.getApplication(),
+                    clientId,
+                    environment,
+                    redirectUrl
+                );
+                PayPalCheckout.setConfig(checkoutConfig);
                 sendPayPalRequest(activity, payPalCheckoutRequest, callback);
             }
         });
 
     }
 
+    private String getPayPalReturnUrl(Context context) {
+        if (context != null) {
+            return String.format("%s://paypalpay", context.getPackageName().toLowerCase(Locale.ROOT));
```

Comment:
We do [strip out underscores](https://github.com/braintree/braintree_android/blob/eb04b3794b7fba185f4dd9a34a8c7242934f09a1/BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.java#L152) when performing a similar operation in `BraintreeClient`, though I can't remember exactly why. Couldn't hurt to do the same here for consitency.

Reply:
Gotcha good to know, I wonder if that was added to guard against package names not matching up with valid application IDs?

I wonder if there are other edge cases that could come up by creating the returnUrl this way (thinking about if an application id doesn't match up with a package). Can definitely dig into this a bit more, stripping the underscores sounds reasonable though 👍